modules/videoheroesBidAdapter.js

```diff
@@ -1,261 +1,286 @@
-import { isEmpty, parseUrl, isStr, triggerPixel } from '../src/utils.js';
-import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { isEmpty, parseUrl, isPlainObject, isArray, isArrayOfNums, isFn, isStr, triggerPixel } from '../src/utils.js';
 import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
 import { config } from '../src/config.js';
 import { convertOrtbRequestToProprietaryNative } from '../src/native.js';
 
-/**
- * @typedef {import('../src/adapters/bidderFactory.js').BidRequest} BidRequest
- * @typedef {import('../src/adapters/bidderFactory.js').Bid} Bid
- */
-
 const BIDDER_CODE = 'videoheroes';
-const DEFAULT_CUR = 'USD';
-const ENDPOINT_URL = `https://point.contextualadv.com/?t=2&partner=hash`;
-
-const NATIVE_ASSETS_IDS = { 1: 'title', 2: 'icon', 3: 'image', 4: 'body', 5: 'sponsoredBy', 6: 'cta' };
-const NATIVE_ASSETS = {
-  title: { id: 1, name: 'title' },
-  icon: { id: 2, type: 1, name: 'img' },
-  image: { id: 3, type: 3, name: 'img' },
-  body: { id: 4, type: 2, name: 'data' },
-  sponsoredBy: { id: 5, type: 1, name: 'data' },
-  cta: { id: 6, type: 12, name: 'data' }
+const MEDIA_TYPES = [BANNER, VIDEO, NATIVE];
+const DEF_FLOOR = 0.05;
+const CUR = 'USD';
+const TTL = 1200;
+const ENDPOINT_URL = `https://point.contextualadv.com/?t=2&partner=`;
+
+const NATIVE_ASSETS_IDS = { 1: 'title', 2: 'icon', 3: 'image', 4: 'sponsoredBy', 5: 'body', 6: 'rating', 7: 'downloads', 8: 'cta' };
+
+const NATIVE_ASSETS_TYPES = {
+  title: {id: 1, title: {len: 25}},
+  icon: {id: 2, img: {type: 1}},
+  image: {id: 3, img: {type: 3}},
+  sponsoredBy: {id: 4, data: {type: 1, len: 30}},
+  body: {id: 5, data: {type: 2, len: 100}},
```

Comment:
It doesn't seem like you've fixed things, instead you've refactored to fool the robot. The opportunity to make the code efficient is just hidden by this commit

Reply:
its not duplication, but it developed by same manner, does it better if i will use switch or if instead ob for per keys ?

Reply:
maybe problem because I developed these connectors before and I still see realisation like that ... ok anyway